### ISSUE TITLE
Promote deployment finalization to stable

### ISSUE THREAD

**Issue by cgwalters** [ostreedev/ostree#3025]
In https://github.com/ostreedev/ostree/pull/1841 we added "locking" for deployment finalization.  It's way past time to promote this to a stable API.

One thing I think would be helpful here is to have the deployment lock change to become a *symlink* to the deployment that's locked.  If that has changed, the lock no longer applies.  This would solve an issue I continually hit:

- `systemctl stop zincati`
- rpm-ostree rebase
- systemctl reboot

And discover the rebase didn't work because we had a lingering lockfile.

Something like `ostree admin unlock-finalize` or so, and perhaps `ostree admin reboot --execute` that would itself invoke `systemctl reboot`.  Incidentally it may be nice to lower the "check for systemd inhibitors" here too.

This is a bit more important now because I'd like to change things to have the deployments be source-of-truth (xref https://github.com/coreos/rpm-ostree/issues/4391 )

Basically, this would make it all much easier to set up a flow (like zincati) which does:

- pull and stage (*and lock*) updates
- opt-in unlock+apply

**Comment by jlebon** [comment]
Might be worth updating the description here now that we have https://github.com/ostreedev/ostree/pull/3077.
